app/scripts/controllers/network/provider-api-tests/shared-tests.js

```diff
@@ -340,88 +389,102 @@ export function testsForRpcMethodAssumingNoBlockParam(method) {
       });
     });
 
-    it(`causes a request to fail with a custom error if the request to Infura returns a ${httpStatus} response 5 times in a row`, async () => {
-      await withMockedInfuraCommunications(async (comms) => {
+    it(`causes a request to fail with a custom error if the request to ${providerType} provider returns a ${httpStatus} response 5 times in a row`, async () => {
+      await withMockedCommunications({ type: providerType }, async (comms) => {
         const request = { method };
 
         // The first time a block-cacheable request is made, the latest block
         // number is retrieved through the block tracker first. It doesn't
         // matter what this is — it's just used as a cache key.
         comms.mockNextBlockTrackerRequest();
-        comms.mockInfuraRpcCall({
+        comms.mockRpcCall({
           request,
           response: {
             error: 'Some error',
             httpStatus,
           },
           times: 5,
         });
-        const promiseForResult = withInfuraClient(
+        comms.mockNextBlockTrackerRequest();
+        const promiseForResult = withClient(
+          { type: providerType },
           async ({ makeRpcCall, clock }) => {
             return await waitForPromiseToBeFulfilledAfterRunningAllTimers(
               makeRpcCall(request),
               clock,
             );
           },
         );
-
-        await expect(promiseForResult).rejects.toThrow(
-          /^InfuraProvider - cannot complete request\. All retries exhausted\..+Gateway timeout/su,
-        );
+        const err =
+          providerType === 'infura'
+            ? infuraClientErrorTimeout
+            : jsonRpcClientErrorTimeout(method);
+        await expect(promiseForResult).rejects.toThrow(err);
       });
     });
   });
 
   ['ETIMEDOUT', 'ECONNRESET', 'SyntaxError'].forEach((errorMessagePrefix) => {
-    it(`retries the request to Infura up to 5 times if an "${errorMessagePrefix}" error is thrown while making the request, returning the successful result if there is one on the 5th try`, async () => {
-      await withMockedInfuraCommunications(async (comms) => {
-        const request = { method };
+    if (
+      providerType === 'infura' ||
```

Comment:
I think it makes sense to add a little something for `ECONNRESET` (even if it might be a good idea to place this knowledge eventually live in `eth-json-rpc-infura`). As for `SyntaxError`, I'm not sure that it is unique to the Infura middleware. It looks like the fetch middleware handles this already, at least in spirit. It [attempts to parse the response body as JSON](https://github.com/MetaMask/eth-json-rpc-middleware/blob/254a66cf5f067f35db96c6a5740c8da9a79e495d/src/fetch.ts#L64-L70), and when that fails (due to a SyntaxError) it will throw an error which is then picked up as a retriable error.

Reply:
Added comment for `ECONNRESET` in [f676e08](https://github.com/MetaMask/metamask-extension/pull/16337/commits/f676e08e27ce08e82ee780eb2712bb792e45735b), will add more comments.

Reply:
Great, thanks!